docs/tutorials/python/mnist.md

```diff
@@ -1,116 +1,135 @@
+
 ## Multilayer Perceptron
 
-We first use [multilayer perceptron](https://en.wikipedia.org/wiki/Multilayer_perceptron) to solve this problem. We
-define a multilayer perceptron by using MXNet's symbolic interface. The
-following command create a place holder variable for the input data.
+The first approach makes use of a [Multilayer Perceptron](https://en.wikipedia.org/wiki/Multilayer_perceptron), MLP for short, to solve this problem. We'll define the MLP using MXNet's symbolic interface. We begin by creating a place holder variable for the input data. When working with an MLP, we need to flatten our 28x28 images into a flat 1-D structure of 784 (28 * 28) raw pixel values. The order of pixel values in the flattened vector does not matter as long as we are being consistent about how we do this across all images.
 
 ```python
 data = mx.sym.var('data')
 # Flatten the data from 4-D shape into 2-D (batch_size, num_channel*width*height)
 data = mx.sym.flatten(data=data)
 ```
+One might wonder if we are discarding valuable information by flattening. That is indeed true and we'll cover this more when we talk about convolutional neural networks where we preserve the input shape. For now, we'll go ahead and work with flattened images.
+
+MLP contains several fully connected layers. A fully connected layer or FC layer for short, is one where each neuron in the layer is connected to every neuron in its preceding layer. From a linear algebra perspective, an FC layer applies an [affine transform](https://en.wikipedia.org/wiki/Affine_transformation) to the *n x m* input matrix *X* and outputs a matrix *Y* of size *n x k*, where *k* is the number of neurons in the FC layer. *k* is also referred to as the hidden size. The output *Y* is computed according to the equation *Y = W X + b*. The FC layer has two learnable parameters, the *m x k* weight matrix *W* and the *m x 1* bias vector *b*.
+
 
-A multilayer perceptron contains several fully-connected layers. A fully-connected
-layer, with an *n x m* input matrix *X* outputs a matrix *Y* with size *n x k*,
-where *k* is often called as the hidden size. This layer has two learnable parameters, the
-*m x k* weight matrix *W* and the *m x 1* bias vector *b*. It compute the
-outputs with *Y = W X + b*.
+In an MLP, the output of most FC layers is fed into an activation function, which applies an element-wise non-linearity. This step is critical and it gives neural networks the ability to classify inputs that are not linearly separable. Common choices for activation functions are sigmoid, tanh, and [rectified linear unit](https://en.wikipedia.org/wiki/Rectifier_%28neural_networks%29) ("ReLU" for short). In this example, we'll use the ReLU activation function which has several desirable properties and is typically considered a default choice.
```

Comment:
Again introduce the acronym "Rectified Linear Unit (ReLU)". No need to say "for short"